itest/rust/src/derive_variant/mod.rs

```diff
@@ -9,7 +9,42 @@ use std::fmt::Debug;
 use crate::itest;
 use godot::bind::FromVariant;
 use godot::bind::ToVariant;
-use godot::builtin::{dict, varray, FromVariant, ToVariant};
+use godot::builtin::{dict, varray, FromVariant, ToVariant, Variant};
+
+#[macro_export]
+macro_rules! roundtrip_with_skip {
+    ($name_to:ident, $name_from:ident, $value:expr, $to_var:expr, $from_var:expr) => {
+        #[itest]
+        fn $name_to() {
+            let s = $value;
+            assert_eq!(s.to_variant(), $to_var.to_variant(),)
+        }
+
+        #[itest]
+        fn $name_from() {
+            assert_eq!(EnumWithSkip::from_variant(&$to_var.to_variant()), $from_var);
+        }
+    };
+}
+#[macro_use]
+mod enums_skip;
+//#[macro_use]
+mod structs_skip;
+
+fn roundtrip<T, U>(value: T, expected: U)
+where
+    T: ToVariant + FromVariant + std::cmp::PartialEq + Debug,
+    U: ToVariant,
+{
+    let expected = expected.to_variant();
+
+    assert_eq!(value.to_variant(), expected, "testing converting to");
+    assert_eq!(
+        value,
+        T::from_variant(&expected),
+        "testing converting back from"
+    );
+}
```

Comment:
This exact function already exists in `variant_test.rs`. Consider moving it one level up to `itest/src/lib.rs` and reuse it.

Reply:
it uses only one argument which offers less assurance in our case but it's still a pretty good test so i changed it for now but didn't squash the commit yet in case it's better with two arguments